Google.Api.Gax.Grpc/Rest/HttpRuleTranscoder.cs

```diff
@@ -131,16 +132,45 @@ static IEnumerable<string> GetValues(FieldDescriptor field, IMessage request)
                 }
 
                 object value = field.Accessor.GetValue(request);
-                if (!field.HasPresence && (value is "" || value is 0.0f || value is 0.0d || value is 0 || value is 0L || value is 0UL || value is UnsignedInt32Zero))
+                if (value is null)
                 {
                     yield break;
                 }
-                yield return value is IFormattable formattable
-                    ? formattable.ToString(format: null, CultureInfo.InvariantCulture)
-                    : value?.ToString();
 
-                // TODO: Handle repeated fields
+                if (!field.HasPresence && !IsRequired(field) && IsDefaultValue(value))
```

Comment:
nit: I think that I would understand this better like: 

```csharp
if (!IsRequired(field) && !field.HasPresence && IsDefaultValue(value))
```

And also, for when `field.HasPresence == true` shouldn't you later check that the field was actually set?

Reply:
Thanks!

And, yes, didn't spot that, sorry!